react/week3/to-do/src/ToDo.js

```diff
@@ -0,0 +1,126 @@
+import React, { Component  } from 'react';
+import {CreatLi} from './CreateLi';
+import {ReadInput} from './ReadInput.jsx'
+import {Edit} from './Edit'
+export class ToDo extends React.Component {
+    constructor(){
+        super();
+        this.editLi = React.createRef();
+        this.state = {
+            list : [{
+                "id": 1,
+                "description": "Get out of bed",
+                "created_date" : "2020-08-11",
+                "edit" : false,
+                "status" : true 
+              },
+              {
+                "id": 2,
+                "description": "Brush teeth",
+                "created_date" : "2020-08-11",
+                "edit" : false,
+                "status" : true 
+              },
+              {
+                "id": 3,
+                "description": "Eat breakfast",
+                "created_date" : "2020-08-11",
+                "edit" : false,
+                "status" : true 
+              }]
+        };
+      this.handelDelet = this.handelDelet.bind(this);
+      this.handelSubmit = this.handelSubmit.bind(this);
+      this.handelEdit = this.handelEdit.bind(this);
+      this.handel = this.handel.bind(this);
+      
+    }     
+  handelDelet(place){
+      const list = this.state.list
+      //list.splice(place,1);
+      list[place].status = null;
+      this.setState({
+          list : list
+      })
+     
+  }
+  handelSubmit(data){
+    const newDo = {"id" : this.state.list.length +1,"description": data.description , "created_date" : data.createdDate ,"status" : true , "edit" : false }
+    const nList = this.state.list;
+    nList.push(newDo);
+    this.setState({
+        list : nList
+    });
+    console.log('this new list ',this.state.list )
+  }
+  handelEdit(data){
+    //const newDo = {"id" : this.state.list.length +1,"description": data.description , "created_date" : data.createdDate ,"status" : true  }
+    const nList = this.state.list;
+    nList[data.index-1].description = data.description;
+    nList[data.index-1].created_date = data.createdDate;
+    nList[data.index-1].edit= false;
+    this.setState({
+        list : nList
+    });
+    console.log(nList);
+    
```

Comment:
Generally, it's not practical to rely on the todo id that correspond to todo index because what if the id wasn't incremental or contained sting?
A better way to determan which todo to update is to compare the new updated todo id with todo id the state

```suggestion
 this.setState(prevState => {
            const newList = prevState.list.map(todo => {
                if (data.index === todo.id) {
                    return {
                        ...todo,
                        description: data.description,
                        created_date: data.createdDate,
                        edit: false,
                    };
                }
                return todo;
            });
            return { list: newList };
        });
```
This approach is much cleaner and you don't have to worry about indexing